nilearn/_utils/plotting.py

```diff
@@ -43,15 +43,15 @@ def generate_design_matrices_figures(
     if design_matrices is None:
         return design_matrices_dict
 
-    for i_run, design_matrix in enumerate(design_matrices, start=1):
+    for i_run, design_matrix in enumerate(design_matrices):
         dmtx_plot = plot_design_matrix(design_matrix)
         dmtx_plot = resize_plot_inches(dmtx_plot, height_change=0.3)
         dmtx_fig = None
         if output:
             # the try is mostly here in case badly formed dict
             try:
                 dmtx_fig = output["design_matrices_dict"][i_run][
-                    "design_matrix"
+                    "design_matrix_png"
```

Comment:
@man-shu 
I test using reverting to svg for all figures and I was getting some really ugly results, so decided to stick to png.

Reply:
OK as long as the plots are readable, I think it should be fine.